src/main/java/uk/gov/hmcts/darts/transcriptions/service/impl/DuplicateRequestDetector.java

```diff
@@ -0,0 +1,50 @@
+package uk.gov.hmcts.darts.transcriptions.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.darts.common.entity.TranscriptionEntity;
+import uk.gov.hmcts.darts.common.entity.TranscriptionTypeEntity;
+import uk.gov.hmcts.darts.common.exception.DartsApiException;
+import uk.gov.hmcts.darts.common.repository.TranscriptionRepository;
+import uk.gov.hmcts.darts.common.repository.TranscriptionTypeRepository;
+import uk.gov.hmcts.darts.transcriptions.exception.TranscriptionApiError;
+import uk.gov.hmcts.darts.transcriptions.model.TranscriptionRequestDetails;
+
+import java.util.Collections;
+import java.util.List;
+
+@Component
+@RequiredArgsConstructor
+public class DuplicateRequestDetector {
+
+    private final TranscriptionRepository transcriptionRepository;
+    private final TranscriptionTypeRepository transcriptionTypeRepository;
+
+    public void checkForDuplicate(TranscriptionRequestDetails requestDetails, boolean isManual) {
+        if (requestDetails.getStartDateTime() != null && requestDetails.getEndDateTime() != null) {
+
+            List<TranscriptionEntity> matchingTranscriptions = transcriptionRepository.findByHearingIdTypeStartAndEnd(
+                requestDetails.getHearingId(),
+                getTranscriptionTypeById(requestDetails.getTranscriptionTypeId()),
+                requestDetails.getStartDateTime(),
+                requestDetails.getEndDateTime()
+            );
+            if (!matchingTranscriptions.isEmpty() && matchesSourceOfTranscriptionRequest(isManual, matchingTranscriptions)) {
+                Integer duplicateTranscriptionId = matchingTranscriptions.get(0).getId();
```

Comment:
is this necessarily correct?
the matchingTranscriptions could contain a manual and a automatic, and the matchesSourceOfTranscriptionRequest just tells you if one of them matches the manual/automatic, but not which one, but you are assuming its the first one returned, which i dont think is necessarily the case?

Reply:
Not sure about this.  We only need 1 match on 'isManual' to declare that a duplicate exists.  We don't care which 1 do we? Just that a match exists.